ls1intum/Artemis
# [Enhancement] Fix minor issues in modal to create/edit a posting

### Checklist

### Motivation and Context


I had to do some re-work on the newly introduced create-edit-modal for posts and answerPosts.

### Description

The improvements are as follows:
- modal is not closed when clicking somewhere outside
- every time a modal is opened for creation, the pre-filled fields exactly represent what is already posted -> the change here is, that if you are about to edit a post, change the text, DISMISS it, and then open the edit-modal again, you will _not_ see the "unsaved" changes, but only the saved ones, e.g. the state of the post shown in the list.
- If you input more characters than allowed, the counter (e.g. 1001/1000) changes to color to red to indicate the length violation.
- The dropdown for tags could not be closed when clicking somewhere else in the create-edit-modal, it only closed on selection
- When markdown formatting was used in the post content, the post style broke, this is fixed now (see image below)
- Some style changes to better distinguish posts visually (see image below)

### Steps for Testing


Repeat it for **Discussion Section at Exercise Page** as well as *Discussion Section at Lecture Page** 

1. Log in to Artemis
2. Navigate to Course Administration of a course with posts enabled
3. Go to an exercise/lecture page, edit a post (must be your own if you are not tutor) and click on X in the modal to dismiss -> see that the changes were not saved and re-opening the modal will _not_ show the unsaved changes
4. Edit a post by using some markdown formatting and save it -> see that it was created successfully and the format looks good
5. Try to put more than 1000 characters in the input field and see that this is not valid
6. Try to submit in invalid form (no title, no content, too much content) -> this should not work
7. Create a new post with tags and try to select a tag from the dropdown, if you want to close the dropdown, you should be able to click anywhere else in the modal and it should disappear 
8. Reassure that creating and editing _answer_posts works as well
9. During the whole process of editing and creating posts and answer posts no related errors should occur in the console

### Review Progress





- Code Review
- Manual Tests
  - [x] Test 1 for Exercise & Lecture discussion section, tested with Post and AnswerPost, edited and created new
  - [x] Test 2 for Exercise & Lecture discussion section, tested with Post and AnswerPost, edited and created new

### Screenshots


This is how the improved modal works:

https://user-images.githubusercontent.com/41366522/130808141-88cac8c1-9e65-494d-ab34-8baac1c029dc.mov

This is difference between BEFORE (left) and AFTER (right) fixing the style for posts using markdown:
<img width="973" alt="Bildschirmfoto 2021-08-25 um 22 26 07" src="https://user-images.githubusercontent.com/41366522/130859982-8ecf72f6-9342-43e7-b4e6-8a6a72cbb91a.png">


This is the difference in the thread listing: BEFORE (left) and AFTER (right) -> thicker separation lines, less padding above "new post"-button
<img width="797" alt="Bildschirmfoto 2021-08-26 um 19 11 05" src="https://user-images.githubusercontent.com/41366522/131006450-75534ebe-b0dd-41b1-ada7-f3138cb9a14f.png">
